x-pack/legacy/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/WatcherFlyout.tsx

```diff
@@ -149,57 +151,32 @@ export class WatcherFlyout extends Component<
   };
 
   public createWatch = () => {
-    const core = this.context;
     const { serviceName } = this.props.urlParams;
 
     if (!serviceName) {
       return;
     }
 
-    const emails = this.state.actions.email
-      ? this.state.emails
-          .split(',')
-          .map(email => email.trim())
-          .filter(email => !!email)
-      : [];
+    const email = this.state.actions.email ? this.state.emails : '';
 
     const slackUrl = this.state.actions.slack ? this.state.slackUrl : '';
 
-    const schedule =
-      this.state.schedule === 'interval'
-        ? {
-            interval: `${this.state.interval.value}${this.state.interval.unit}`
-          }
-        : {
-            daily: { at: `${this.state.daily}` }
-          };
+    const timeRange = {
+      value: this.state.interval.value,
+      unit: this.state.interval.unit
+    };
 
-    const timeRange =
-      this.state.schedule === 'interval'
-        ? {
-            value: this.state.interval.value,
-            unit: this.state.interval.unit
-          }
-        : {
-            value: 24,
-            unit: 'h'
-          };
-
-    const apmIndexPatternTitle = core.injectedMetadata.getInjectedVar(
-      'apmIndexPatternTitle'
-    ) as string;
-
-    return createErrorGroupWatch({
-      emails,
-      schedule,
+    return createErrorOccurrenceAlert({
+      callApmApi: this.props.callApmApi,
+      email,
       serviceName,
       slackUrl,
       threshold: this.state.threshold,
-      timeRange,
-      apmIndexPatternTitle
+      timeRange
     })
-      .then((id: string) => {
+      .then(savedObject => {
         this.props.onClose();
+        const id = 'id' in savedObject ? savedObject.id : NOT_AVAILABLE_LABEL;
```

Comment:
Maybe I'm outmoded, but isn't the `in` operator generally discouraged since it's at risk for prototype hijacking?